keps/sig-storage/1698-generic-ephemeral-volumes/README.md

```diff
@@ -580,3 +601,35 @@ The advantage is that no automatically created PVCs are
 needed. However, other controllers also create user-visible objects
 (statefulset -> pod and PVC, deployment -> replicaset -> pod), so this
 concept is familiar to users.
+
+### Extending CSI ephemeral volumes
+
+In the current CSI ephemeral volume design, the CSI driver only gets
+involved after a pod has already been scheduled onto a node. For feature
+parity with normal volumes, the CSI driver would have to reimplement
+the work done by external-provisioner and external-attacher. This makes
+CSI drivers specific to Kubernetes, which is against the philosophy of
+CSI.
+
+For storage capacity aware pod scheduling, the Kubernetes API would
+have to be extended to make the scheduler aware of the ephemeral
+volume's size. Kubelet would need to be extended to evict a pod when
+volume creation fails because storage is exhausted. Currently it just
+retries until someone or something else deletes the pod.
+
+All of that would lead to additional complexity both in Kubernetes and in
+CSI drivers. Handling volume provisioning and attaching through the
+existing mechanisms is easier and compatible with future enhancements
+of those.
```

Comment:
I think I will challenge here this assumption that CNI, CSI, or CRI should be independent of Kube.  They exist to serve the needs of the Kube community.

The point of ephemeral volumes is to delegate to the node to manage node local volumes.  If you're saying the point of this change is to move every single volume interaciton through a PVC, that's a much bigger statement than (maybe i want persistent volumes sometimes).

Reply:
I.e. both local storage and attached devices fit within the PVC model.  Are you proposing that things that do not need any form of resource control, scheduling, or constraint, should also go through that model?

If so this proposal is really "Rename PersistentVolume and PersistentVolumeClaim to be Volume and VolumeClaim".

Reply:
@thockin would you agree or disagree with that interpretation?

Reply:
If we could rename these API types, then yes, renaming them would be ideal. The KEP doesn't include such a proposal because cleaning up the names is rather disruptive with no real gain other than avoiding potential confusion. But perhaps that's worth it? I'm open for suggestions here.